src/context_free.ml

```diff
@@ -597,53 +608,56 @@ class map_top_down ?(expect_mismatch_handler = Expect_mismatch_handler.nop)
                         item.pstr_loc (Many items);
                     items @ loop rest ~in_generated_code)
             | _ -> (
-                let item = super#structure_item base_ctxt item in
-                match item.pstr_desc with
-                | Pstr_type (rf, tds) ->
+                let expanded_item = super#structure_item base_ctxt item in
+                match (item.pstr_desc, expanded_item.pstr_desc) with
+                | Pstr_type (rf, tds), Pstr_type (exp_rf, exp_tds) ->
+                    assert (Poly.(rf = exp_rf));
```

Comment:
It was originally. I don't think it's necessary now because this can't actually be rewritten but this acts as a failsafe in the event that we ever change that.

I can add a comment or simply remove it, let me know what you prefer!

Reply:
To me it sounds good to keep it. Thanks for the explanation!